### PR TITLE
Backend: performance improvement trevor trapper

### PR DESCRIPTION
**Pull request by hannibal002**
## What
performance improvement trevor trapper

exclude_from_changelog

### CODE REVIEW

File: src/main/java/at/hannibal2/skyhanni/features/misc/trevor/TrevorFeatures.kt [hunk 1]
```diff
@@ -323,6 +324,11 @@ object TrevorFeatures {
         resetTrapper()
     }
 
+    @SubscribeEvent
+    fun onTick(event: LorenzTickEvent) {
+        inTrapperDen = LorenzUtils.skyBlockArea == "Trapper's Den"
```
Comment (j10a1n15) [inline src/main/java/at/hannibal2/skyhanni/features/misc/trevor/TrevorFeatures.kt]: Probably update it on scoreboard change + use repopattern

**Review by CalMWolfs — CHANGES_REQUESTED** [review]
j10a1n15's comment

File: src/main/java/at/hannibal2/skyhanni/features/misc/trevor/TrevorFeatures.kt (hunk 1 above)
Comment (hannibal002) [reply src/main/java/at/hannibal2/skyhanni/features/misc/trevor/TrevorFeatures.kt]: repo pattern is good.
scorboard change event is tricky: we either need to delay the LorenzUtils.skyBlockArea check by 1-2 ticks, or we need to check the scoreboard data directly. both is not ideal.

Comment (CalMWolfs) [reply src/main/java/at/hannibal2/skyhanni/features/misc/trevor/TrevorFeatures.kt]: we can do the scoreboarch change later

**Review by CalMWolfs — APPROVED** [review]

**Pull request merged**